src/renderer/src/lib/modal/InstallFromFile.svelte

```diff
@@ -37,15 +34,20 @@
 			},
 			{
 				onSuccess: () => {
-					$installedApps.refetch();
 					const filePath = files ? files[0].path : '';
+					$installedApps.refetch().then((apps) => {
+						// Only open the app if it's an app with UI
+						if (filePath.endsWith('.webhapp')) {
+							const newApp = apps.data?.find(
+								(extendedAppInfo) => extendedAppInfo.appInfo.installed_app_id === formData.appId
+							);
+							if (newApp) $openApp.mutate(newApp);
+						}
+					});
 					toastStore.trigger({
 						message: $i18n.t('appInstalled'),
 						background: 'variant-filled-success'
 					});
-					if (filePath.endsWith('.webhapp')) {
```

Comment:
This was the old logic to navigate to the launcher view with the search field prepopulated